### PR TITLE
Convert unit tests to googletest framework

### PR DESCRIPTION
**Pull request by danielhlarkin**
The goal of this PR is to improve compile times for our unittests. The Catch framework, while easy to use, can be very slow to compile with optimizations, ASAN, etc. enabled, especially with long tests with many variables. Google's test framework proves to be much quicker (up to 10x).

### CODE REVIEW

File: tests/Agency/CleanUpLostCollectionTest.cpp [hunk 1]
```diff
@@ -125,54 +132,60 @@ SECTION("clean up a lost collection when the leader is failed") {
     //        old: /arango/Supervision/Health/leader/Status == "FAILED"
 
     auto const& trxs = q->slice();
-    REQUIRE(trxs.length() == 1);
+    EXPECT_TRUE(trxs.length() == 1);
```
Comment (gnusi) [inline tests/Agency/CleanUpLostCollectionTest.cpp]: `REQUIRE` in catch == `ASSERT_TRUE` in gtest

Comment (danielhlarkin) [reply tests/Agency/CleanUpLostCollectionTest.cpp]: True, but the `ASSERT_` methods issue a `return;` and so cannot be used in methods with a non-void return type.

Comment (gnusi) [reply tests/Agency/CleanUpLostCollectionTest.cpp]: got it

**Comment by gnusi** [comment]
wouldn't be simpler to just define corresponding macros without modifying most of the tests? e.g.
```
#define REQUIRE(x) ASSERT_TRUE(x)
```

**Comment by danielhlarkin** [comment]
@gnusi If the macros were 1-to-1 transformable, sure. But as found above, some instances of `REQUIRE` need to be `EXPECT_TRUE` instead of `ASSERT_TRUE`, while most should still be transformed. Either way, it's already done, so no real reason to go back and undo the changes.

Going forward, we would probably want to stick to the Google-y assertions anwyay.

**Comment by gnusi** [comment]
@danielhlarkin absolutely, just curious :-)

**Comment by danielhlarkin** [comment]
Did a quick compilation timing test with a `RelWIthDebInfo` build, `make -j6`.

catch 10m18s
gtest 6m53s

Less dramatic than I was hoping, but non-trivial. I'll try to get some ASAN numbers from one of the cluster machines on Monday (not enough RAM to build it on my local machine).

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/4529/

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/4554/
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-mac-onlycatch/2531/

File: CMakeLists.txt [hunk 2]
```diff
@@ -764,9 +764,9 @@ if (USE_MAINTAINER_MODE)
   find_program(AWK_EXECUTABLE awk)
 endif ()
 
-option(USE_CATCH_TESTS "Compile catch C++ tests" ON)
-if (USE_CATCH_TESTS)
-  add_definitions("-DARANGODB_USE_CATCH_TESTS=1")
+option(USE_GOOGLE_TESTS "Compile catch C++ tests" ON)
```
Comment (jsteemann) [inline CMakeLists.txt]:
```suggestion
option(USE_GOOGLE_TESTS "Compile C++ unit tests" ON)
```

**Pull request merged**